### PR TITLE
Emails: update inbox textual information to encourage trial users to use more titan accounts

### PR DESCRIPTION
**Pull request by nightnei**
#### Proposed Changes

This pull request updates the "Create new mailbox" upsell element to encourage new mailbox creation during Titan's free trial.
So the users see a ribbon with "free" text and a new description. And other users just see another new description to have consistency.

#### Testing Instructions
Let's check a few scenarios.
**1) Test appearing of the ribbon and the new description of the upsell element (*for Professional Email*)**

* Go to your dashboard
* Click on the "Switch Site" button in the top left corner
* If you have existed account with Professional Email (FREE TRIAL):
	* Choose your account with the Professional Email (FREE TRIAL)
* Otherwise, if you don't have existed account with the Professional Email (FREE TRIAL):
    * Click on "Add new site" at the end of the appeared list
    * Enter any domain name for test purposes
    * In the appeared list below - choose a domain with the help of the "Select" button
    * Choose any plan from the proposed with the help of the "Select" button
    * Scroll down and pay for the subscription with the help of the "Pay {amount} with credits" button
    * Fill data in the "Professional Email" form to create your 3-month free email account and click on the "Add professional Email" button
    * Scroll down and click on "Complete Checkout"
    * On the page "What are your goals?" - click on "Skip to dashboard" in the top right corner
* In the left sidebar menu, click on the "Inbox" item

| BEFORE  | AFTER |
| ------------- | ------------- |
| ------------- | We can see the ribbon with "free" text and the new description|
image | image

**2) Test that the ribbon has disappeared for non-trial users, and the description of the upsell element has been changed (*for Professional Email*)**
* Move the mouse on the "Upgrades" item in the left sidebar
* In the dropdown, click on the "Emails" item
* Click on the "Renew now" button, which is located at the top part of the page body (on the right side of the text "Expires: {Date}")
* Scroll down and click on "Pay {amount} with credits"
* In the top left corner of the page, click the "My Sites" button
* In the left sidebar, click on the "Inbox" item

| BEFORE  | AFTER |
| ------------- | ------------- |
| ------------- | Ribbon disappeared, and now we have another new description |
image | image

**3) Test that in case of buying "Google Workspace" - we see only new description and we never see the ribbon**

* Go to your dashboard
* Click on the "Switch Site" button in the top left corner
* Recommended if you have existed account with "Google Workspace" subscription:
	* Choose your account with the "Google Workspace" subscription
* Otherwise, if you don't have existed account with the "Google Workspace" subscription:
    * Click on "Add new site" at the end of the appeared list
    * Enter any domain name for test purposes
    * In the appeared list below - choose a domain with the help of the "Select" button
    * Choose any plan from the proposed with the help of the "Select" button
    * Scroll down and pay for the subscription with the help of the "Pay {amount} with credits" button
    * Note, on the page "Add Professional Email" - click on the "Skip for now" button
    * On the page "What are your goals?" - click on "Skip to dashboard" in the top right corner
    * In the left sidebar menu, click on the "Inbox" item
    * At the bottom of the page is located the "Google Workspace" section, click on the "Select" button, on the right side of the section
    * Fill "Google Workspace" form and click on the "Purchase" button, below the form
    * Click on "Pay {amount} with credits"
    * Click on the "Manage email" button in the middle of the page body
    * In the left sidebar menu, click on the "Inbox" item
    * If you see the warning "Configuring mailboxes" at the top part of the page body - it means that it's necessary to wait a bit, so at the current moment, you can go to the kitchen and make a cup of coffee 🙃
    * Reload the page to check for any news regarding the configuration of your mailboxes. 
    * If it's finished, you will see the notification "Action required". So click on the "Finish setup" button under the notification and follow the instructions.
	* After finishing setup - come back to the dashboard
* In the left sidebar, click on the "Inbox" item

| BEFORE  | AFTER |
| ------------- | ------------- |
| ------------- | We see the same result as for paid "Professional Email" |
image | image

#### Pre-merge Checklist
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [x] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [x] Have you checked for TypeScript, React or other console errors?
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?

Related to #
1200182182542585-as-1202735679392334/f

### CODE REVIEW

File: client/my-sites/email/inbox/new-mailbox-upsell/index.tsx [hunk 1]
```diff
@@ -22,13 +22,17 @@ const NewMailboxUpsell = ( { domains }: { domains: ResponseDomain[] } ) => {
...
+
 	// User has one single domain, determine / email addition page URL based on subscribed email provider.
 	if ( domains.length === 1 ) {
 		const domainItem = domains[ 0 ];
 
 		let slug = '';
 		if ( hasTitanMailWithUs( domainItem ) ) {
 			slug = 'titan/new';
+
+			isTrialNow = isUserOnTitanTrial( domainItem );
```
Comment (wongasy) [inline client/my-sites/email/inbox/new-mailbox-upsell/index.tsx]: @nightnei We probably don't need to introduce this new function here. We can rely on [getTitanMailboxPurchaseCost( domainItem )](https://github.com/Automattic/wp-calypso/blob/38c37bbfe35b8c741c4e346637987ddc0fb3ab59/client/lib/titan/get-titan-mailbox-purchase-cost.js) to retrieve the purchase cost object, we can set the `isTrialNow` flag by checking if `getTitanMailboxPurchaseCost( domainItem ).amount` is `0`

@AllTerrainDeveloper Please correct me if I'm wrong on this one.

File: client/my-sites/email/inbox/new-mailbox-upsell/index.tsx [hunk 2]
```diff
@@ -43,9 +47,16 @@ const NewMailboxUpsell = ( { domains }: { domains: ResponseDomain[] } ) => {
 	return (
 		<div className="new-mailbox-upsell__container">
 			<div className="new-mailbox-upsell">
+				{ isTrialNow ? <Ribbon color="green">FREE</Ribbon> : null }
```
Comment (wongasy) [inline client/my-sites/email/inbox/new-mailbox-upsell/index.tsx]: We can probably do the shorthand `isTrialNow ?? <Ribbon color="green">FREE</Ribbon>` here :)

**Review by wongasy — CHANGES_REQUESTED** [review]

File: client/my-sites/email/inbox/new-mailbox-upsell/index.tsx (hunk 2 above)
Comment (nightnei) [reply client/my-sites/email/inbox/new-mailbox-upsell/index.tsx]: Good point, thank you :)
Only instead of nullish coalescing it's necessary to use logical &&, since the first one renders element only in case of null/undefined.

[Article with recommendation from React](https://reactjs.org/docs/conditional-rendering.html#inline-if-with-logical--operator)

File: client/my-sites/email/inbox/new-mailbox-upsell/index.tsx (hunk 1 above)
Comment (nightnei) [reply client/my-sites/email/inbox/new-mailbox-upsell/index.tsx]: It will really work, and a similar approach [was already used in two other places](https://github.com/Automattic/wp-calypso/pull/66567/files#diff-faaeae7cc3937862cc4b14995597ad17f60733a2168b82f2d60ed59ebefe35c8L87).
But from my point of view - it's not obvious that `amount === 0` means that it's a free trial, and for me, it looks like some type of hardcode that is copy-pasted in a few places. That's why I decided to create the function, to isolate comparison and make the codebase more readable and clear (I spent some time finding and being sure that `amount === 0` is indeed the correct comparison to check the free trial).

What do you think about it? @wongasy @AllTerrainDeveloper

File: client/my-sites/email/inbox/new-mailbox-upsell/index.tsx (hunk 2 above)
Comment (stephanethomas) [reply client/my-sites/email/inbox/new-mailbox-upsell/index.tsx]: I think we usually follow this approach for conditional blocks:

```suggestion
				{ isTrialNow && <Ribbon color="green">FREE</Ribbon> }
```

File: client/my-sites/email/inbox/new-mailbox-upsell/index.tsx (hunk 1 above)
Comment (stephanethomas) [reply client/my-sites/email/inbox/new-mailbox-upsell/index.tsx]: Even better, if you wait a bit @olaseni has added selectors in https://github.com/Automattic/wp-calypso/pull/66153. So you will be able to call `isDomainEligibleForTitanFreeTrial()` here 😀.

Comment (nightnei) [reply client/my-sites/email/inbox/new-mailbox-upsell/index.tsx]: @stephanethomas I have checked @olaseni changes and `isDomainEligibleForTitanFreeTrial` do another things.
`isDomainEligibleForTitanFreeTrial` - means that a user doesn't have Professional email yet, but they are allowed to use free trial.
And my function `isUserOnTitanTrial ` - means that user already took Professional email and now they are on free trial.

File: client/my-sites/email/inbox/new-mailbox-upsell/index.tsx (hunk 2 above)
Comment (nightnei) [reply client/my-sites/email/inbox/new-mailbox-upsell/index.tsx]: @stephanethomas, thank you, you are right. As I already mentioned - it's recommended way by React.
[Link for documentation](https://reactjs.org/docs/conditional-rendering.html#inline-if-with-logical--operator).

File: client/my-sites/email/inbox/new-mailbox-upsell/index.tsx (hunk 1 above)
Comment (stephanethomas) [reply client/my-sites/email/inbox/new-mailbox-upsell/index.tsx]: Ah right, good point. I was just passing by but I should definitively have paid more attention 😅.

May I suggest renaming `isUserOnTitanTrial()` to `isUserOnTitanFreeTrial()` though? That way it's consistent with `isDomainEligibleForTitanFreeTrial()` and the rest of our code.

File: client/my-sites/email/inbox/new-mailbox-upsell/index.tsx (hunk 2 above)
Comment (AllTerrainDeveloper) [reply client/my-sites/email/inbox/new-mailbox-upsell/index.tsx]: I was discussing this with @nightnei in our onboarding session. II agree that is better to use a logical && coalesce. Our code is plenty of `condition && <Element/>` and probably the best thing to do here is basically to change it like this:
```suggestion
				{ isTrialNow && <Ribbon color="green">FREE</Ribbon> }
```

File: client/my-sites/email/inbox/new-mailbox-upsell/index.tsx (hunk 1 above)
Comment (AllTerrainDeveloper) [reply client/my-sites/email/inbox/new-mailbox-upsell/index.tsx]: @wongasy I feel your concern, but I think it's a good addition to our code base. I imagine myself in the situation of having to check some user accounts being on trial or not, and I just found it difficult and tricky. I think adding a new function will ease further developments.

The only thing I would propose is to split the refactor in two pull request. As I discussed with Vova in our onboarding session, breaking down the pull requests as much as possible would benefit from:

- Faster reviews
- Faster testing
- Faster deploys
- More secure changes
- Better tracking

Comment (nightnei) [reply client/my-sites/email/inbox/new-mailbox-upsell/index.tsx]: > Ah right, good point. I was just passing by but I should definitively have paid more attention 😅.

np 😉

> May I suggest renaming isUserOnTitanTrial() to isUserOnTitanFreeTrial() though? That way it's consistent with isDomainEligibleForTitanFreeTrial() and the rest of our code.

Totally agree with you, and actually, I have already thought about it too and done it locally 😄

**Comment by AllTerrainDeveloper** [comment]
@nightnei good job creating this pull request as your first one in Automattic! 🎉

File: client/lib/titan/test/lib/fixtures.ts [hunk 3]
```diff
@@ -0,0 +1,114 @@
+import type { ResponseDomain, EmailCost } from 'calypso/lib/domains/types';
+
+const exampleDomain: ResponseDomain = {
```
Comment (fredrikekelund) [inline client/lib/titan/test/lib/fixtures.ts]: This is also just a drive-by observation, but could we maybe use `DOMAIN_PRIMARY` from `client/state/sites/domains/test/fixture/index.js` as the base object here instead?

I haven't inspected any potential differences between the two, but it's another dummy instance of `ResponseDomain`, also used for testing.

Comment (nightnei) [reply client/lib/titan/test/lib/fixtures.ts]: Wow, I haven't found it previously, thank you ❤️

**Comment by wongasy** [comment]
@nightnei Making great progress :) As discussed earlier, let's also wait until we have the tracking in place before deploying this feature, so we can effectively track the impact of this feature!

File: client/my-sites/email/inbox/new-mailbox-upsell/index.tsx [hunk 4]
```diff
@@ -43,9 +47,16 @@ const NewMailboxUpsell = ( { domains }: { domains: ResponseDomain[] } ) => {
...
 		<div className="new-mailbox-upsell__container">
 			<div className="new-mailbox-upsell">
+				{ isFreeTrialNow && <Ribbon color="green">FREE</Ribbon> }
 				<div className="new-mailbox-upsell__messages">
 					<h2>{ translate( 'Need another mailbox?' ) }</h2>
-					<div>{ translate( 'Create a new one and activate it immediately.' ) }</div>
+					<div>
+						{ isFreeTrialNow
+							? translate(
+									'Create a new one free during your trial to experience multiple mailbox efficiency.'
```
Comment (wongasy) [inline client/my-sites/email/inbox/new-mailbox-upsell/index.tsx]: This is what we had in the asana card, but reading the sentence over again, it might sounds smoother with a `for` before `free`

>Create a new one `for` free during your trial to experience multiple mailbox efficiency.

Comment (nightnei) [reply client/my-sites/email/inbox/new-mailbox-upsell/index.tsx]: Done

**Review by wongasy — APPROVED** [review]
LGTM

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/7493757

Thank you @nightnei for including a screenshot in the description! This is really helpful for our translators.

**Comment by wongasy** [comment]
@nightnei I found a new test case that we should have tested. The current code works fine in this new case, so we're all good, but I just wanted to flag this case so we remember to test it next time!

When the site has multiple domains (email subscriptions), we should not show the ribbon:

Screen Shot 2022-08-24 at 1 59 27 PM

If we really want, we could show the ribbon if all email subscriptions are free trials, but I think it's fine to leave the feature as-is for the time being and see how it performs.

**Comment by nightnei** [comment]
> @nightnei I found a new test case that we should have tested. The current code works fine in this new case, so we're all good, but I just wanted to flag this case so we remember to test it next time!
> 
> When the site has multiple domains (email subscriptions), we should not show the ribbon:
> 
> Screen Shot 2022-08-24 at 1 59 27 PM
> 
> If we really want, we could show the ribbon if all email subscriptions are free trials, but I think it's fine to leave the feature as-is for the time being and see how it performs.

Wow, really, I will keep it in mind 👍

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/7493757).
